airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/IntegrationCommand.kt

```diff
@@ -0,0 +1,165 @@
+        val result = execute(command)
+        result.onSuccess { airbyteMessage ->
+            airbyteMessage?.let { outputRecordCollector.accept(airbyteMessage) }
+        }
+        result.onFailure {
+            commandSpec.commandLine().usage(System.out)
+            logger.error(it) { "\nUnable to perform operation '$command'." }
+            // Many of the exceptions thrown are nested inside layers of RuntimeExceptions. An
+            // attempt is made
+            // to
+            // find the root exception that corresponds to a configuration error. If that does not
+            // exist, we
+            // just return the original exception.
+            ApmTraceUtils.addExceptionToTrace(it)
+            val rootThrowable = ConnectorExceptionUtil.getRootConfigError(Exception(it))
+            val displayMessage = ConnectorExceptionUtil.getDisplayMessage(rootThrowable)
+            // If the source connector throws a config error, a trace message with the relevant
+            // message should
+            // be surfaced.
+            if (ConnectorExceptionUtil.isConfigError(rootThrowable)) {
+                AirbyteTraceMessageUtility.emitConfigErrorTrace(it, displayMessage)
+            }
+            if (OperationType.CHECK.name.equals(command, true)) {
+                // Currently, special handling is required for the CHECK case since the user display
+                // information in
+                // the trace message is
+                // not properly surfaced to the FE. In the future, we can remove this and just throw
+                // an exception.
+                outputRecordCollector.accept(
+                    AirbyteMessage()
+                        .withType(AirbyteMessage.Type.CONNECTION_STATUS)
+                        .withConnectionStatus(
+                            AirbyteConnectionStatus()
+                                .withStatus(AirbyteConnectionStatus.Status.FAILED)
+                                .withMessage(displayMessage),
+                        ),
+                )
+            }
+        }
+
+        logger.info { "Completed integration: $connectorName" }
+    }
+
+    private fun execute(command: String): Result<AirbyteMessage?> {
+        return try {
+            val operationType = OperationType.valueOf(command.uppercase())
+            val operation: Operation? =
+                operations.firstOrNull { o: Operation -> (operationType == o.type()) }
+            operation?.execute()
+                ?: Result.failure(
+                    IllegalArgumentException(
+                        "Connector does not support the '${operationType.name.lowercase()}' operation.",
+                    ),
+                )
+        } catch (e: IllegalArgumentException) {
+            Result.failure(
+                IllegalArgumentException(
+                    "Connector does not support the '${command.lowercase()}' operation.",
+                    e
+                )
```

Comment:
You might be wrapping an IllegalArgumentException inside another with this.

Reply:
@postamar Good catch.  The intent is to catch the failure and put a meaningful exception in the failure result.  I can create a new exception type for this instead of using `IllegalArgumentException` (like `IntegrationCommandException`).  Thoughts?

Reply:
This does sound like it might make for nicer assertThrows in tests, too

Reply:
Done